src/platform/api/pythonApi.ts

```diff
@@ -80,20 +79,34 @@ export function pythonEnvToJupyterEnv(env: ResolvedEnvironment): PythonEnvironme
             envType = EnvironmentType.VirtualEnv;
         }
     }
+    let isCondaEnvWithoutPython = false;
+    let uri: Uri;
+    let id = env.id;
     if (!env.executable.uri) {
-        traceWarning(`Python environment ${env.id} excluded as Uri is undefined`);
-        return;
+        if (envType === EnvironmentType.Conda && supportsEmptyCondaEnv) {
+            isCondaEnvWithoutPython = true;
+            uri =
+                getOSType() === OSType.Windows
+                    ? Uri.joinPath(env.environment?.folderUri || Uri.file(env.path), 'python.exe')
+                    : Uri.joinPath(env.environment?.folderUri || Uri.file(env.path), 'bin', 'python');
+        } else {
+            traceError(`Python environment ${env.id} excluded as Uri is undefined`);
+            return;
+        }
+    } else {
+        uri = env.executable.uri;
     }
 
     return {
-        id: env.id,
+        id,
         sysPrefix: env.executable.sysPrefix || '',
         envPath: env.environment?.folderUri,
         displayPath: env.environment?.folderUri || Uri.file(env.path),
         envName: env.environment?.name || '',
-        uri: env.executable.uri,
+        uri,
         displayName: env.environment?.name || '',
         envType,
+        isCondaEnvWithoutPython,
```

Comment:
This partially brings in the support for #11855, but not fully implemented an not displayed
We do not display Conda envs without Python, however internally in the API (code) we detect them as we need to look for non-python kernels such as R kernels that are in conda environments.

Hence the reason to partially support #11855 (but not yet fully support it)